java/src/jmri/jmrix/AbstractProgrammer.java

```diff
@@ -125,11 +126,22 @@ public boolean getCanRead(String addr) {
     }
 
     // handle mode
-    protected ProgrammingMode mode = DefaultProgrammerManager.PAGEMODE;
+    private ProgrammingMode mode = DefaultProgrammerManager.PAGEMODE;
```

Comment:
I'm not sure if this conflicts with [line 165](https://github.com/JMRI/JMRI/pull/541/files#diff-2af1f6574126f0f6b4b31b3300a41dddR165)